client/src/main/scala/com.crobox.clickhouse/stream/ClickhouseSink.scala

```diff
@@ -12,10 +12,22 @@ import scala.concurrent.{ExecutionContext, Future}
 
 case class ClickhouseIndexingException(msg: String, cause: Throwable, payload: Seq[String], table: String)
     extends RuntimeException(msg, cause)
-case class Insert(table: String, jsonRow: String)
+
+sealed trait TableOperation {
+  def table: String
+}
+
+case class Insert(table: String, jsonRow: String) extends TableOperation
+
+case class Optimize(table: String, cluster: Option[String]) extends TableOperation
 
 object ClickhouseSink extends LazyLogging {
 
+  /**
+   * @deprecated added support to optimize table operations <br/>
+   *             use [[#toSink()]] instead.
+   */
+  @deprecated
```

Comment:
@deprecated("use [[#toSink()]] instead")

Reply:
Also,  I was referring to:

def insertSink(config: Config, client: ClickhouseClient, indexerName: Option[String] = None)(
      implicit ec: ExecutionContext,
      settings: QuerySettings = QuerySettings()
  ): Sink[Insert, Future[Done]] = toSink(config, client, indexerName)